### PR TITLE
Api16 minimum

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description
The custom tabs dependency requires API16 going forward, and #5623 means API15 clients won't be able to sync after 1/1/2020 anyway.

Google statistics shows there are only about 300 people left on API15, so moving forward one notch is minimum pain

## Approach
Standard compat infrastructure adjustment as APIs move up:
- mainline any code from CompatV16 to the API consumers (as the API is available everywhere now)
- remove all CompatV16 APIs from Compat and CompatV16 and CompatV15
- move all remaining CompatV15 implementations to CompatV16
- adjust CompatHelper to return CompatV16 by default, and remove anything unused

## How Has This Been Tested?

- re-run all tests on full emulator skew locally (API16 plus a sample of others up to API29)
- watch Travis to make sure it doesn't blow up (Travis runs every API possible)

## Learning (optional, can help others)

https://android.googlesource.com/platform/frameworks/support/+/6e847a4064cdda740b778d560b1db477e13e0772/browser/build.gradle#13

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/libanki/HttpTest.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/CompatV16.java  -1
+ AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/libanki/DBTest.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/timrae/Anki-Android/pullRequest?prid=4708610&bid=15728164)

File: AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java [hunk 1]
```diff
@@ -72,11 +74,44 @@ public void onDisabled(Context context) {
...
+            Bundle options = manager.getAppWidgetOptions(id);
+            float width, height;
+            if (context.getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
+                width = options.getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_WIDTH);
+                height = options.getInt(AppWidgetManager.OPTION_APPWIDGET_MAX_HEIGHT);
+            } else {
+                width = options.getInt(AppWidgetManager.OPTION_APPWIDGET_MAX_WIDTH);
+                height = options.getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_HEIGHT);
+            }
+            int horizontal, vertical;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/app/timrae/Anki-Android/pullRequest?prid=4708610)

File: AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java [hunk 2]
```diff
@@ -72,11 +74,44 @@ public void onDisabled(Context context) {
...
         super.onReceive(context, intent);
     }
 
+    private static void updateWidgetDimensions(Context context, RemoteViews updateViews, Class<?> cls) {
+        AppWidgetManager manager = AppWidgetManager.getInstance(context);
+        int[] ids = manager.getAppWidgetIds(new ComponentName(context, cls));
+        for (int id : ids) {
+            final float scale = context.getResources().getDisplayMetrics().density;
+            Bundle options = manager.getAppWidgetOptions(id);
+            float width, height;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/app/timrae/Anki-Android/pullRequest?prid=4708610)

File: AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java]: was in original implementation, this was just copy/paste as I inlined the compat code

File: AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java (hunk 2 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java]: was in original implementation, this was just copy/paste as I inlined the compat code

**Comment by mikehardy** [comment]
2.10 is nearly ready for release and I prefer to delay API cutoffs as long as possible so I will likely release a 2.10 stable then target this for 2.11, if anyone is watching

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/compat/CompatV16.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java  -3
+ AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/libanki/DBTest.java  -1
+ AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/libanki/HttpTest.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4708610&bid=15728164)

File: AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java (hunk 1 above)
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java:96]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4708610)

File: AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java (hunk 2 above)
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/widget/AnkiDroidWidgetSmall.java:88]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4708610)

**Pull request merged**